### ISSUE TITLE
Improve pretty-printing: wrap long lines, and abbreviate long lists and deep structures

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#15]
Improve value formatting. Currently, given
```
val emps =
  [{id = 100, name = "Fred", deptno = 10},
   {id = 101, name = "Velma", deptno = 20},
   {id = 102, name = "Shaggy", deptno = 30},
   {id = 103, name = "Scooby", deptno = 30}];
```
the shell will print
```
val emps =
  [{deptno=10,id=100,name="Fred"},{deptno=20,id=101,name="Velma"},
   {deptno=30,id=102,name="Shaggy"},{deptno=30,id=103,name="Scooby"}] : {deptno:int, id:int, name:string} list
```
but sml would print
```
val it =
  [{deptno=10,id=100,name="Fred"},{deptno=20,id=101,name="Velma"},
   {deptno=30,id=102,name="Shaggy"},{deptno=30,id=103,name="Scooby"}]
  : {deptno:int, id:int, name:string} list
```
and we should emulate that. Note how ': type' are on a new line.

**Comment by julianhyde** [comment]
Also, abbreviate long lists and deeply nested structures:
* If lists have more than 12 entries, write '...' for the remaining entries.
* In deeply nested structures, print '#' for elements more than 5 deep.

**Comment by julianhyde** [comment]
Fixed in cf2e8da.
